### PR TITLE
Add a data source for reading attributes from a servers certificate chain

### LINKED ISSUE #4: Proposal: New data source for TLS cert
_This issue was originally opened by @martinssipenko as hashicorp/terraform#14856. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

I would like to propose a new data source under `TLS` provider named `tls_cert` (or something like that).

Arguments:
- You would be able to either supply a path to TLS certificate as argument, or supply a PEM, which could be read by `${file('path/to/cert.pem')}`.

Attributes:
- `signature_algo`
- `public_key_algo`
- `serial_number` (represented as integer)
- `serial_number_hex` (represented as hex delimited with `:`)
- `is_ca`
- `version`
- `issuer`
- `subject`
- `not_before`
- `not_after`

Possibly even more.

Please share your view on introducing such provider.

### PR DESCRIPTION
**Pull request by wjam**
Fixes #4 and #52.

### CODE REVIEW

**Comment by wjam** [comment]
@appilon @apparentlymart @radeksimko Is this okay to be reviewed?

**Comment by wjam** [comment]
@apparentlymart @radeksimko @appilon Is this change okay to be reviewed/merged?

**Comment by wjam** [comment]
@apparentlymart @radeksimko @appilon Any update on review this PR?

**Comment by wjam** [comment]
@apparentlymart @radeksimko @appilon Is there any chances of getting this reviewed?

**Comment by rabidscorpio** [comment]
This would allow a completely automated setup of an EKS cluster without the need to run an external script and allow https://github.com/terraform-providers/terraform-provider-aws/issues/10104 to be closed, an issue that's been around for 8 months now. @apparentlymart @radeksimko @appilon Please take a look at this.

**Comment by asmfis** [comment]
hello, wonder if there is any update on this? do you need help with approving / reviewing it @apparentlymart @radeksimko @appilon?

File: tls/test_fixtures/tls_certs/private.pem [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (kmoe) [inline tls/test_fixtures/tls_certs/private.pem:1]: Rename `test_fixtures` directory to `testdata` following Go convention.

**Comment by wjam** [comment]
@kmoe Changes pushed

**Pull request merged**